### PR TITLE
mgr/cephadm: add --no-exception-when-missing flag to cert-store cert/key get

### PR DESCRIPTION
**Pull request by adk3798**
This is mostly intended for automation that wants to check for these certificates and handle the results themselves. Without this flag an exception will be logged which isn't great for automation tools that may check for this repeatedly

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -3264,10 +3264,13 @@ def cert_store_get_cert(
         self,
         entity: str,
         service_name: Optional[str] = None,
-        hostname: Optional[str] = None
+        hostname: Optional[str] = None,
+        no_exception_when_missing: bool = False
```
Comment (nizamial09) [inline src/pybind/mgr/cephadm/module.py:3268]: i went for `ignore_missing_exception` in dashboard btw.

**Review by nizamial09 — APPROVED** [review]
lgtm, thank you!

**Comment by nizamial09** [comment]
jenkins test api

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-09-21_16:57:11-orch:cephadm-wip-adk-testing-2024-09-21-1014-distro-default-smithi/

Failures:
- 6x mds_upgrade_sequence timeouts, known issue
- 3x SELinux denials + stray daemon warnings, known issue
- 2x rgw-ingress timeout, known issue
- test_nfs task failed due to PR in the run
- staggered upgrade with agent enabled failed, known issue
- stray daemon warning in rados_python task (not a big issue, needs an ignorelist entry)

Nothing to block merging outside of the PR causing the test_nfs task failure

**Comment by adk3798** [comment]
jenkins test api

**Pull request merged**